### PR TITLE
Some code refactor for 'exec_plan_fragment'

### PR DESCRIPTION
**Pull request by xlwh**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Some code refactor for 'exec_plan_fragment'.  The original logic has not been changed. Thanks!

### CODE REVIEW

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -411,35 +420,24 @@ Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, co
...
-            return Status::InternalError("Double execute");
-        }
-        // register exec_state before starting exec thread
-        _fragment_map.insert(std::make_pair(fragment_instance_id, exec_state));
+    // Check fragment exec whether exist again.
+    if (IsFragmentExecExist(fragment_instance_id)) {
+        return Status::InternalError("Double execute");
     }
+    // Register exec_state before starting exec thread.
+    _fragment_map.insert(std::make_pair(fragment_instance_id, exec_state));
```
Comment (trueeyu) [inline be/src/runtime/fragment_mgr.cpp:434]: fragment_map insert without lock, may be dangerous

Comment (xlwh) [reply be/src/runtime/fragment_mgr.cpp]: OK, thanks.

File: be/src/runtime/fragment_mgr.cpp [hunk 2]
```diff
@@ -411,35 +420,27 @@ Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, co
...
+            params.params.query_id, fragment_instance_id, params.backend_num, _exec_env, params.coord);
     RETURN_IF_ERROR_WITH_WARN(exec_state->prepare(params), "Fail to prepare Fragment");
-
+    // Check fragment exec whether exist again.
+    if (IsFragmentExecExist(fragment_instance_id)) {
+        return Status::InternalError("Double execute");
+    }
+    // Register exec_state before starting exec thread.
     {
         std::lock_guard<std::mutex> lock(_lock);
```
Comment (trueeyu) [inline be/src/runtime/fragment_mgr.cpp:435]: There is a problem with this change. The previous logic was to insert it if it was not found. But after the change, lock are twice. 
The following facts will happen: In the first step, it is found that it does not exist. But when the second step is executed, it already exists.

Comment (xlwh) [reply be/src/runtime/fragment_mgr.cpp:435]: Ok, you are right, this refactor may not be good. thanks.

**Pull request closed without merging**